src/System/HardwareInput.h

```diff
@@ -0,0 +1,78 @@
+#ifndef _HARDWARE_INPUT_H_
+#define _HARDWARE_INPUT_H_
+
+#include <stdint.h>
+#include <functional>
+#include "settings.h"
+
+typedef int32_t input_data_t;
+typedef std::function<void(input_data_t)> InputCallback;
+
+class HardwareInput {
+    public:
+        virtual ~HardwareInput() { }
+        virtual void init() = 0;
+        virtual void run() = 0;
+        void setAction(InputCallback action) { _action = action; }
+
+    protected:
+        InputCallback _action = [](int32_t) { }; // default action
+};
+
+class HardwareDemuxer {
```

Comment:
would you be able to add a comment explaining what this class does? Its not immediately clear to me

Reply:
For sure.  Have you worked with multiplexers before?  They allow you to split up signals from one to many.  We were running low on pins for all the different functions of dyno, so i added multiplexer for all the ui button inputs which allows us to read 4 buttons while only using 3 pins

Reply:
I actually wound up removing that class from this PR.  I've got another PR ready to go for when this one is done which concerns reading the demuxxed inputs

Reply:
Ahh okay. So the multiplexer would be a hardware component that is represented by this HardwareDemuxer class?

Reply:
exactly